src/main/java/pl/debuguj/parkingspacessystem/dao/ParkingSpaceDaoMockImpl.java

```diff
@@ -0,0 +1,117 @@
+        return Collections.unmodifiableList(listParkingSpaces);
+    }
+
+    @Override
+    public Optional<ParkingSpace> updateEndTime(
+            final String registrationNumber,
+            final Date timestamp) throws IncorrectEndDateException, ParkingSpaceNotFoundException {
+
+        Optional<ParkingSpace> ps = findByRegistrationNo(registrationNumber);
+
+        if(ps.isPresent())
+        {
+            if (!timestamp.after(ps.get().getBeginTime())) {
+                throw new IncorrectEndDateException("Incorrect update time");
+            }
+
+            listParkingSpaces
+                    .forEach(s -> {
+                        if (Objects.equals(registrationNumber, s.getCarRegistrationNumber())) {
+                            s.setEndTime(timestamp);
+                        }
+                    });
+
+            return findByRegistrationNo(registrationNumber);
+        }
+        throw new ParkingSpaceNotFoundException("Parking space not found");
+    }
+
+    @Override
+    public void removeAll() {
+        listParkingSpaces.clear();
+    }
+
+    @Override
+    public List<ParkingSpace> findByDate(final Date timestamp) {
+        final Date end = createEndDate(timestamp);
+        return Collections.unmodifiableList(
+                listParkingSpaces
+                .stream()
+                .filter(ps -> timestamp.before(ps.getBeginTime())
+                        && end.after(ps.getBeginTime()))
+                .collect(Collectors.toList()));
+
+    }
+
+    private Date createEndDate(final Date d){
+        return (new DateTime(d).plusDays(1)).toDate();
+    }
+
+//    @Override
+//    public void onApplicationEvent(ApplicationEvent applicationEvent) {
+//        try {
+//            listParkingSpaces.clear();
+//
+//            listParkingSpaces.create(new ParkingSpace("11111",
+//                    timeDateFormat.parse("2017-10-13 10:25:48"),
+//                    timeDateFormat.parse("2017-10-13 10:35:12")));
+//            listParkingSpaces.create(new ParkingSpace("22222",
+//                    timeDateFormat.parse("2017-10-13 12:25:48"),
+//                    timeDateFormat.parse("2017-10-13 13:35:12")));
+//            listParkingSpaces.create(new ParkingSpace("33333",
+//                    timeDateFormat.parse("2017-10-13 15:25:48"),
+//                    timeDateFormat.parse("2017-10-13 16:35:12")));
+//            listParkingSpaces.create(new ParkingSpace("44444",
+//                    timeDateFormat.parse("2017-10-14 20:25:48"),
+//                    timeDateFormat.parse("2017-10-14 21:35:12")));
+//            listParkingSpaces.create(new ParkingSpace("55555",
+//                    timeDateFormat.parse("2017-10-14 11:15:48"),
+//                    timeDateFormat.parse("2017-10-14 12:35:12")));
+//        } catch (ParseException e){
+//            e.printStackTrace();
+//        } catch (IncorrectEndDateException e) {
+//            e.printStackTrace();
+//        }
+//    }
```

Comment:
a to co?

Reply:
Na końcu przejrzałem kod, ale widocznie można było zrobić to dokładniej.